### PR TITLE
[SYCL][E2E] Refactor `check_device_code/group_[load|store].cpp` tests

### PR DESCRIPTION
**Pull request by aelovikov-intel**
So that `CHECK`s could be re-generated without manual intervention.

### CODE REVIEW

**Comment by aelovikov-intel** [comment]
To reviewers: I've uploaded this PR as **two** commits. The final state is fully auto-generated `CHECK`s, but the diff is hard to follow. As such, I've introduced an intermediate commit that manually repositions chunks of code to resemble the old version more.

I think reviewing this on per-commit bases will be much easier than do it all at once.

**Comment by aelovikov-intel** [comment]
@intel/llvm-reviewers-runtime , Chris is OOO, can someone else take a look please? It should make downstream pulldowns much smoother.

**Comment by jsji** [comment]
> @intel/llvm-reviewers-runtime , Chris is OOO, can someone else take a look please? It should make downstream pulldowns much smoother.

Thanks @aelovikov-intel for improving the maintainability !

File: sycl/test/check_device_code/group_load.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --functions "group_load" --include-generated-funcs --version 4
+// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py
 // NOTE: and manually adjusted to follow the related explicit instantiation.
```
Comment (jsji) [inline sycl/test/check_device_code/group_load.cpp:2]: Should we remove this note if CHECKs are now *fully automated*

**Review by uditagarwal97 — APPROVED** [review]
